server/remote_cache/hit_tracker/hit_tracker.go

```diff
@@ -331,11 +331,34 @@ func cacheEventTypeLabel(c counterType) string {
 	return uploadLabel
 }
 
-func sizeMetric(ct counterType) *prometheus.HistogramVec {
+func emitSizeMetrics(ct counterType, cacheTypeLabel, serverLabel string, digestSizeBytes, bytesTransferredCache, bytesTransferredClient float64) {
 	if ct == UploadSizeBytes {
-		return metrics.CacheUploadSizeBytes
-	}
-	return metrics.CacheDownloadSizeBytes
+		metrics.CacheUploadSizeBytes.With(prometheus.Labels{
+			metrics.CacheTypeLabel: cacheTypeLabel,
+			metrics.ServerName:     serverLabel,
+		}).Observe(bytesTransferredCache)
+		metrics.ServerUploadSizeBytes.With(prometheus.Labels{
+			metrics.CacheTypeLabel: cacheTypeLabel,
+			metrics.ServerName:     serverLabel,
+		}).Observe(bytesTransferredClient)
+		metrics.DigestUploadSizeBytes.With(prometheus.Labels{
+			metrics.CacheTypeLabel: cacheTypeLabel,
+			metrics.ServerName:     serverLabel,
+		}).Observe(digestSizeBytes)
```

Comment:
missing return

Reply:
This function doesn't return anything (before it returned the name of the metric, now it actually emits the metrics)